contracts/crosschain/ERC7802Bridge.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.27;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IERC20, SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {BitMaps} from "@openzeppelin/contracts/utils/structs/BitMaps.sol";
+import {Bytes} from "@openzeppelin/contracts/utils/Bytes.sol";
+import {Pausable} from "@openzeppelin/contracts/utils/Pausable.sol";
+import {InteroperableAddress} from "@openzeppelin/contracts/utils/draft-InteroperableAddress.sol";
+import {IERC7786GatewaySource, IERC7786Receiver} from "../interfaces/IERC7786.sol";
+import {IERC7802} from "../interfaces/IERC7802.sol";
+
+abstract contract ERC7802Bridge is IERC7786Receiver, Ownable, Pausable {
+    using BitMaps for BitMaps.BitMap;
+    using InteroperableAddress for bytes;
+
+    event Sent(address token, address from, bytes to, uint256 amount);
+    event Received(address token, bytes from, address to, uint256 amount);
+
+    event GatewayRegistered(address indexed gateway, bytes chain);
+    event RemoteBridgeRegistered(address indexed token, bytes chain);
+
+    error ERC7802BridgeMissingGateway(bytes chain);
+    error ERC7802BridgeMissingRemote(address token, bytes chain);
+    error ERC7802BridgeImproperChainIdentifier(bytes chain);
+    error ERC7802BridgeDuplicate();
+    error ERC7802BridgeInvalidGateway();
+    error ERC7802BridgeInvalidSender();
+
+    mapping(address token => mapping(bytes chain => bytes bridge)) private _remoteBridges;
+    mapping(address token => mapping(bytes chain => bytes remote)) private _remoteTokens;
+    mapping(bytes chain => address gateway) private _gateways;
+    BitMaps.BitMap private _processed;
+
+    function getGateway(bytes memory chain) public view virtual returns (address gateway) {
+        gateway = _gateways[chain];
+        require(gateway != address(0), ERC7802BridgeMissingGateway(chain));
+    }
+
+    function getRemoteBridge(address token, bytes memory chain) public view virtual returns (bytes memory bridge) {
+        bridge = _remoteBridges[token][chain];
+        require(bridge.length > 0, ERC7802BridgeMissingRemote(token, chain));
+    }
+
+    function getRemoteToken(address token, bytes memory chain) public view virtual returns (bytes memory remote) {
+        remote = _remoteTokens[token][chain];
+        require(remote.length > 0, ERC7802BridgeMissingRemote(token, chain));
+    }
+
+    // TODO: check that gateway is not address(0) ? prevent  override ?
```

Comment:
If we're checking that gateway is not `address(0)` here I don't think we should check it on the `getGateway`. I'm generally unsure whether to check `address(0)` in the first place.